Realm/RLMSchema.mm

```diff
@@ -71,66 +92,92 @@ + (void)initialize {
+
+            s_localNameToClass[className] = cls;
+
+            RLMReplaceClassNameMethod(cls, className);
+            // override sharedSchema classs method to return nil to avoid topo-sort issues
+            RLMReplaceSharedSchemaMethod(cls, nil);
         }
 
-        NSString *className = NSStringFromClass(cls);
-        if ([RLMSwiftSupport isSwiftClassName:className]) {
-            className = [RLMSwiftSupport demangleClassName:className];
-        }
-        // NSStringFromClass demangles the names for top-level Swift classes
-        // but not for nested classes. _T indicates it's a Swift symbol, t
-        // indicates it's a type, and C indicates it's a class.
-        else if ([className hasPrefix:@"_TtC"]) {
-            NSString *message = [NSString stringWithFormat:@"RLMObject subclasses cannot be nested within other declarations. Please move %@ to global scope.", className];
-            @throw RLMException(message);
-        }
+        NSMutableArray *schemas = [NSMutableArray arrayWithCapacity:s_localNameToClass.count];
+        for (Class cls in s_localNameToClass.allValues) {
+            RLMObjectSchema *schema = [RLMObjectSchema schemaForObjectClass:cls];
 
-        if (s_localNameToClass[className]) {
-            NSString *message = [NSString stringWithFormat:@"RLMObject subclasses with the same name cannot be included twice in the same target. Please make sure '%@' is only linked once to your current target.", className];
-            @throw RLMException(message);
+            // set standalone class on shared shema for standalone object creation
+            schema.standaloneClass = RLMStandaloneAccessorClassForObjectClass(schema.objectClass, schema);
+
+            // override sharedSchema classs methods for performance
+            RLMReplaceSharedSchemaMethod(cls, schema);
+
+            if ([cls shouldIncludeInDefaultSchema]) {
+                [schemas addObject:schema];
+            }
         }
-        s_localNameToClass[className] = cls;
 
-        // override classname for all valid classes
-        RLMReplaceClassNameMethod(cls, className);
+        // protected by the @synchronized around s_localNameToClass
+        s_partialSharedSchema.objectSchema = [[s_partialSharedSchema objectSchema] arrayByAddingObjectsFromArray:schemas] ?: schemas;
     }
+}
 
-    // process all RLMObject subclasses
-    for (Class cls in s_localNameToClass.allValues) {
-        RLMObjectSchema *schema = [RLMObjectSchema schemaForObjectClass:cls];
-        [schemaArray addObject:schema];
+// schema based on runtime objects
++ (instancetype)sharedSchema {
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [&](){
+        RLMSchema *schema = [[RLMSchema alloc] init];
 
-        // override sharedSchema classs methods for performance
-        RLMReplaceSharedSchemaMethod(cls, schema);
+        unsigned int numClasses;
+        std::unique_ptr<__unsafe_unretained Class[]> classes(objc_copyClassList(&numClasses));
```

Comment:
This will use `delete[]` when it needs to use `free()`. `std::unique_ptr<__unsafe_unretained Class[], void (*)(void *)> classes(objc_copyClassList(&numClasses), &free);`